fairseq/models/fconv.py

```diff
@@ -494,8 +494,13 @@ def get_normalized_probs(self, net_output, log_probs, sample):
         """Get normalized probabilities (or log probs) from a net's output."""
 
         if self.adaptive_softmax is not None:
-            assert sample is not None and 'target' in sample
-            out = self.adaptive_softmax.get_log_prob(net_output[0], sample['target'])
+            if sample is not None:
+                # Assert enforced only at training time
+                assert 'target' in sample
+                out = self.adaptive_softmax.get_log_prob(net_output[0], sample['target'])
+            else:
+                # No target was provided, setting to None
+                out = self.adaptive_softmax.get_log_prob(net_output[0], target=None)
```

Comment:
can we move "target" out into a variable and then we can just call get_log_prob once unconditionally? (to make refactors easier in the future)